frontend/src/pages/modelRegistry/screens/RegisterModel/utils.ts

```diff
@@ -100,29 +163,100 @@ export const registerVersion = async (
+        string_value: formData.modelSourceGroup,
+      };
+    }
+
+    if (formData.modelSourceId) {
+      artifactCustomProperties.modelSourceId = {
+        metadataType: ModelRegistryMetadataType.STRING,
+        // eslint-disable-next-line camelcase
+        string_value: formData.modelSourceId,
+      };
+    }
+
+    if (formData.modelSourceName) {
+      artifactCustomProperties.modelSourceName = {
+        metadataType: ModelRegistryMetadataType.STRING,
+        // eslint-disable-next-line camelcase
+        string_value: formData.modelSourceName,
+      };
+    }
+
+    // For backward compatibility, set catalog custom properties
+    if (formData.catalogSourceName) {
+      artifactCustomProperties._registeredFromCatalogSourceName = {
+        metadataType: ModelRegistryMetadataType.STRING,
+        // eslint-disable-next-line camelcase
+        string_value: formData.catalogSourceName,
+      };
+    }
+
+    if (formData.catalogRepositoryName) {
+      artifactCustomProperties._registeredFromCatalogRepositoryName = {
+        metadataType: ModelRegistryMetadataType.STRING,
+        // eslint-disable-next-line camelcase
+        string_value: formData.catalogRepositoryName,
+      };
+    }
+
+    if (formData.catalogModelName) {
+      artifactCustomProperties._registeredFromCatalogModelName = {
+        metadataType: ModelRegistryMetadataType.STRING,
+        // eslint-disable-next-line camelcase
+        string_value: formData.catalogModelName,
+      };
+    }
+
+    if (formData.catalogModelTag) {
+      artifactCustomProperties._registeredFromCatalogTag = {
+        metadataType: ModelRegistryMetadataType.STRING,
+        // eslint-disable-next-line camelcase
+        string_value: formData.catalogModelTag,
+      };
+    }
+
+    const artifactData: CreateModelArtifactData = {
       name: `${formData.versionName}`,
-      description: formData.versionDescription,
-      customProperties: {},
+      description: formData.versionDescription || '',
+      customProperties: artifactCustomProperties,
       state: ModelArtifactState.LIVE,
       author,
       modelFormatName: formData.sourceModelFormat,
       modelFormatVersion: formData.sourceModelFormatVersion,
-      // TODO fill in the name of the data connection we used to prefill if we used one
-      // TODO this should be done as part of https://issues.redhat.com/browse/RHOAIENG-9914
-      // TODO should be fixed via https://issues.redhat.com/browse/RHOAIENG-19921
-      // storageKey: 'TODO',
+      // Set top-level pipeline source properties
+      modelSourceGroup: formData.modelSourceGroup,
+      modelSourceId: formData.modelSourceId,
+      modelSourceName: formData.modelSourceName,
+      // Set top-level catalog properties
+      catalogSourceName: formData.catalogSourceName,
+      catalogRepositoryName: formData.catalogRepositoryName,
+      catalogModelName: formData.catalogModelName,
+      catalogModelTag: formData.catalogModelTag,
```

Comment:
These don't need to be separated by pipeline and catalog, we'll simply include the `modelSource*` properties - and make sure to include them all including `modelSourceKind` and `modelSourceClass`, refer to what's added in [the backend PR](https://github.com/kubeflow/model-registry/pull/838/files#diff-76e892700d0c69e6da9e12e5f92028b640e18a02d359d3e7fc4a32854a10dc9d).

If we're bundling up `additionalArtifactProperties` in formData as I mentioned, we can remove all these lines and instead just spread that here:

```ts
...(formData.additionalArtifactProperties || {})
```